lib/msg-format.c

```diff
@@ -124,12 +124,25 @@ msg_format_parse(MsgFormatOptions *options, LogMessage *msg, const guchar *data,
 
   msg_format_preprocess_message(options, msg, data, length);
 
+  if (!msg_format_process_message(options, msg, data, length, problem_position))
+    return FALSE;
+
+  msg_format_postprocess_message(options, msg, data, length);
+  return TRUE;
+}
+
+void
+msg_format_parse(MsgFormatOptions *options, LogMessage *msg, const guchar *data, gsize length)
+{
   gsize problem_position = 0;
 
-  if (!msg_format_process_message(options, msg, data, length, &problem_position))
-    msg_format_inject_parse_error(msg, data, _rstripped_message_length(data, length), problem_position);
+  if (!msg_format_parse_conditional(options, msg, data, length, &problem_position))
+    {
+      msg_format_inject_parse_error(msg, data, _rstripped_message_length(data, length), problem_position);
 
-  msg_format_postprocess_message(options, msg, data, length);
+      /* the injected error message needs to be postprocessed too */
+      msg_format_postprocess_message(options, msg, data, length);
```

Comment:
Why do we need to post-process messages that are failed to parse?
Isn't processing `no-multi-line` could interfere with the `>@<` mark from the error-injecting function?

I'm okay that `msg_format_parse_conditional` does post-processing, as the message will be passed over for further processing in the log path.

Reply:
The same reason as above, failed parsing results in a message too.